metrics.py

```diff
@@ -346,7 +346,6 @@ class ExWeightNumEdges(Metric):
     def __init__(self, exogenous_attr: Collection):
         super().__init__(requires_graph=False)
         self.exogenous_attr = exogenous_attr
-        self.num_weight_mats = self._get_num_weight_mats()
```

Comment:
Wasn't it better to have this as an attribute which is set at initialization, and can be reused?
still not sure why we want to call get_num_weight_mats() every time (but maybe I'll find out by the end of the CR)

Reply:
It was like that at the beginning, but during the fixing (and potentially any time) the value should change as `indices_to_ignore` is updated, so it is safer to calculate it every time with the current `indices_to_ignore`